bin/generate_matrix.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+set -e
+
+# # gives the path of the dockerfiles that have been modified/added in the last commit to main
+changed_dockerfile_path=$(git diff --name-only $1 $2 | grep '/Dockerfile$')
+
+# Count the number of file paths
+total_files=$(echo "${changed_dockerfile_path}" | wc -l)
+
+# i dont exactly know why we write include in the json that we are creating for the matrix but if it works it works
+# if someone knows let me know. Thanks
+echo -n '{\"include\": ['
```

Comment:
The `include` key is used for specifying individual entries to the overall matrix ([see docs](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations)). This is useful as it allows setting up a dictionary of key/values that are different per entry, and not needing to worry about how they would combine if you were using matrix variables.